src/helpers/GiveRedemptionBalanceHelper.ts

```diff
@@ -97,50 +114,60 @@ export const getMockRedemptionBalancesAsync = async ({ address, networkID, provi
-  const zeroPadAddress = ethers.utils.hexZeroPad(address, 32);
-
-  const givingContract = new ethers.Contract(addresses[networkID].GIVING_ADDRESS as string, OlympusGiving, provider);
-
-  // creates a filter looking at all Deposited events on the YieldDirector contract
-  const filter = {
-    address: addresses[networkID].GIVING_ADDRESS,
-    fromBlock: 1,
-    toBlock: "latest",
-    topics: [ethers.utils.id("Deposited(address,address,uint256)"), null, zeroPadAddress], // hash identifying Deposited event
-  };
+  const donationsToAddress = [];
+
+  if (!(addresses[networkID] && addresses[networkID].GIVING_ADDRESS)) {
+    console.log("Unable to find MOCK_SOHM contract on chain ID " + networkID);
+    return;
+  } else {
+    // Addresses in EVM events are zero padded out to 32 characters and are lower case
+    // This matches our inputs with the data we expect to receive from Ethereum
+    const zeroPadAddress = ethers.utils.hexZeroPad(address, 32);
 
-  // using the filter, get all events
-  const events = await provider.getLogs(filter);
+    const givingContract = new ethers.Contract(addresses[networkID].GIVING_ADDRESS as string, OlympusGiving, provider);
 
-  const donorAddresses: IDonorAddresses = {};
-  const donationsToAddress = [];
-  for (let i = 0; i < events.length; i++) {
-    const event = events[i];
-
-    if (event.topics[2] === zeroPadAddress.toLowerCase()) {
-      const donorActiveDonations: [string[], BigNumber[]] = await givingContract.getAllDeposits(
-        ethers.utils.hexDataSlice(event.topics[1], 12),
-      );
-      // make sure the deposit was an active donation and has not been withdrawn
-      for (let j = 0; j < donorActiveDonations[0].length; j++) {
-        // Makes sure that the recipient matches the one we are looking for,
-        // that the deposit amount is not 0, and that the user is not already
-        // counted for donating in a previous Deposited event
-        if (
-          donorActiveDonations[0][j].toLowerCase() == address.toLowerCase() &&
-          donorActiveDonations[1][j] > BigNumber.from(0) &&
-          !donorAddresses[event.topics[1]]
-        ) {
-          donationsToAddress.push(event);
-          // keep track of active donors so multiple deposits are not counted as multiple donors
-          donorAddresses[event.topics[1]] = true;
+    // creates a filter looking at all Deposited events on the YieldDirector contract
+    const filter = {
+      address: addresses[networkID].GIVING_ADDRESS,
+      fromBlock: 1,
+      toBlock: "latest",
+      topics: [ethers.utils.id("Deposited(address,address,uint256)"), null, zeroPadAddress], // hash identifying Deposited event
+    };
+
+    // using the filter, get all events
+    const events = await provider.getLogs(filter);
+
+    const donorAddresses: IDonorAddresses = {};
+    for (let i = 0; i < events.length; i++) {
+      const event = events[i];
+
+      if (event.topics[2] === zeroPadAddress.toLowerCase()) {
```

Comment:
Easier to read if you check whether a condition isn't satisfied, and `continue` if so.